cypress/integration/collections_spec.js

```diff
@@ -0,0 +1,123 @@
+const ace = require('brace');
+require('brace/mode/javascript');
+require('brace/theme/monokai');
+
+
+import { shouldBeRedirectedToLogin } from '../support/assertions';
+
+describe('Dashboard Collections Page', () => {
+  describe('When not logged in', () => {
+    it('should redirect to login page', () => {
+      cy.visit('/#/collections');
+      shouldBeRedirectedToLogin();
+    });
+  });
+
+  describe('When logged in', () => {
+    const name = 'TESTCOLLECTION';
+    const version = '006';
+    beforeEach(() => {
+      cy.login();
+    });
+
+    after(() => {
+      cy.task('resetState');
+    });
+
+    it('displays a link to view collections', () => {
+      cy.visit('/');
+
+      cy.contains('nav li a', 'Collections').should('exist').as('collections');
+      cy.get('@collections').should('have.attr', 'href', '#/collections');
+      cy.get('@collections').click();
+
+      cy.url().should('include', 'collections');
+      cy.get('.heading--xlarge').should('have.text', 'Collections');
+
+      cy.get('table tbody tr').its('length').should('be.eq', 5);
+    });
+
+    it('collections page displays a button to add a new collection', () => {
+      cy.visit('/#/collections');
+
+      cy.get('.heading--large').should('have.text', 'Collection Overview');
+      cy.contains('a', 'Add a Collection').should('exist').as('addCollection');
+      cy.get('@addCollection').should('have.attr', 'href', '#/collections/add');
+      cy.get('@addCollection').click();
+
+      // fill the form and submit
+      const collection = '{{}"name":"TESTCOLLECTION","version":"006","dataType":"TESTCOLLECTION", "duplicateHandling":"error"}';
+      cy.get('textarea').type(collection, {force: true});
+      cy.get('form').get('input').contains('Submit').click();
+
+      // displays the new collection
+      cy.get('.heading--xlarge').should('have.text', 'Collections');
+      cy.get('.heading--large').should('have.text', `${name} / ${version}`);
+      cy.url().should('include', `#/collections/collection/${name}/${version}`);
+
+      // verify the new collection is added to the collections list
+      cy.contains('a', 'Back to Collections').click();
+      cy.contains('table tbody tr a', name)
+        .should('exist')
+        .and('have.attr', 'href', `#/collections/collection/${name}/${version}`);
+    });
+
+    it('collection page has button to edit the collection', () => {
+      cy.visit(`/#/collections/collection/${name}/${version}`);
+      cy.contains('a', 'Edit').should('exist').as('editCollection');
+      cy.get('@editCollection')
+        .should('have.attr', 'href')
+        .and('include', `#/collections/edit/${name}/${version}`);
+      cy.get('@editCollection').click();
+
+      cy.get('.heading--large').should('have.text', `Edit ${name}___${version}`);
+
+      // There are issues updating the react form which uses ace editor.
```

Comment:
I'm not sure we need to include these comments. People can just reference the actually working approach here